The failures are related.
```
[08:33:06]W:     [Step 2/3] [ERROR] /store/work/tc-work/bb15431f347cd651/testsuite/integration/basic/src/test/java/org/jboss/as/test/integration/messaging/mgmt/ExternalConnectionFactoryManagementTestCase.java:35:15: Unused import - org.jboss.as.test.integration.messaging.jms.external.DiscoveryGroupExternalMessagingDeploymentTestCase.REMOTE_PCF. [UnusedImports]
[08:33:06]W:     [Step 2/3] [ERROR] /store/work/tc-work/bb15431f347cd651/testsuite/integration/basic/src/test/java/org/jboss/as/test/integration/messaging/mgmt/ExternalConnectionFactoryManagementTestCase.java:49:8: Unused import - org.jboss.as.controller.client.helpers.ClientConstants. [UnusedImports]
[08:33:06]W:     [Step 2/3] [ERROR] /store/work/tc-work/bb15431f347cd651/testsuite/integration/basic/src/test/java/org/jboss/as/test/integration/messaging/mgmt/ExternalConnectionFactoryManagementTestCase.java:52:8: Unused import - org.jboss.as.test.integration.common.jms.JMSOperationsException. [UnusedImports]
```

--- a/messaging-activemq/src/main/java/org/wildfly/extension/messaging/activemq/MessagingServices.java
+++ b/messaging-activemq/src/main/java/org/wildfly/extension/messaging/activemq/MessagingServices.java
@@ -23,7 +23,9 @@
 package org.wildfly.extension.messaging.activemq;
 
 import static org.wildfly.extension.messaging.activemq.CommonAttributes.JMS_BRIDGE;
+import static org.wildfly.extension.messaging.activemq.CommonAttributes.SUBSYSTEM;
 
+import org.jboss.as.controller.OperationContext;
 import org.jboss.as.controller.PathAddress;
 import org.jboss.as.controller.PathElement;
 import org.jboss.msc.service.ServiceName;
@@ -88,4 +90,7 @@ public class MessagingServices {
        return JBOSS_MESSAGING_ACTIVEMQ.append(JMS_BRIDGE).append(bridgeName);
    }
 
+    public static boolean isSubsystemResource(final OperationContext context) {
+        return context.getCurrentAddress().size() > 0 && SUBSYSTEM.equals(context.getCurrentAddress().getParent().getLastElement().getKey());
+    }
 }

--- a/messaging-activemq/src/main/java/org/wildfly/extension/messaging/activemq/jms/ConnectionFactoryRemove.java
+++ b/messaging-activemq/src/main/java/org/wildfly/extension/messaging/activemq/jms/ConnectionFactoryRemove.java
@@ -22,6 +22,8 @@
 
 package org.wildfly.extension.messaging.activemq.jms;
 
+import static org.wildfly.extension.messaging.activemq.MessagingServices.isSubsystemResource;
+
 import org.jboss.as.controller.AbstractRemoveStepHandler;
 import org.jboss.as.controller.OperationContext;
 import org.jboss.as.controller.OperationFailedException;
@@ -40,12 +42,18 @@ public class ConnectionFactoryRemove extends AbstractRemoveStepHandler {
 
     public static final ConnectionFactoryRemove INSTANCE = new ConnectionFactoryRemove();
 
+    @Override
     protected void performRuntime(OperationContext context, ModelNode operation, ModelNode model) throws OperationFailedException {
-        final String name = context.getCurrentAddressValue();
-        final ServiceName serviceName = MessagingServices.getActiveMQServiceName(context.getCurrentAddress());
-        context.removeService(JMSServices.getConnectionFactoryBaseServiceName(serviceName).append(name));
+        ServiceName serviceName;
+        if(isSubsystemResource(context)) {
+            serviceName = MessagingServices.getActiveMQServiceName("");
+        } else {
+            serviceName = MessagingServices.getActiveMQServiceName(context.getCurrentAddress());
+        }
+        context.removeService(JMSServices.getConnectionFactoryBaseServiceName(serviceName).append(context.getCurrentAddressValue()));
     }
 
+    @Override
     protected void recoverServices(OperationContext context, ModelNode operation, ModelNode model) throws OperationFailedException {
         ConnectionFactoryAdd.INSTANCE.performRuntime(context, operation, model);
     }

--- a/messaging-activemq/src/main/java/org/wildfly/extension/messaging/activemq/jms/PooledConnectionFactoryAdd.java
+++ b/messaging-activemq/src/main/java/org/wildfly/extension/messaging/activemq/jms/PooledConnectionFactoryAdd.java
@@ -28,8 +28,8 @@ import static org.wildfly.extension.messaging.activemq.CommonAttributes.LOCAL;
 import static org.wildfly.extension.messaging.activemq.CommonAttributes.LOCAL_TX;
 import static org.wildfly.extension.messaging.activemq.CommonAttributes.NONE;
 import static org.wildfly.extension.messaging.activemq.CommonAttributes.NO_TX;
-import static org.wildfly.extension.messaging.activemq.CommonAttributes.SUBSYSTEM;
 import static org.wildfly.extension.messaging.activemq.CommonAttributes.XA_TX;
+import static org.wildfly.extension.messaging.activemq.MessagingServices.isSubsystemResource;
 import static org.wildfly.extension.messaging.activemq.jms.ConnectionFactoryAttribute.getDefinitions;
 
 import java.util.ArrayList;
@@ -166,10 +166,6 @@ public class PooledConnectionFactoryAdd extends AbstractAddStepHandler {
 
             installStatistics(context, name);
         }
-    }
-
-    private static boolean isSubsystemResource(final OperationContext context) {
-        return SUBSYSTEM.equals(context.getCurrentAddress().getParent().getLastElement().getKey());
     }
 
     static String getDiscoveryGroup(final ModelNode model) {

--- a/messaging-activemq/src/main/java/org/wildfly/extension/messaging/activemq/jms/PooledConnectionFactoryRemove.java
+++ b/messaging-activemq/src/main/java/org/wildfly/extension/messaging/activemq/jms/PooledConnectionFactoryRemove.java
@@ -22,15 +22,13 @@
 
 package org.wildfly.extension.messaging.activemq.jms;
 
-import static org.jboss.as.controller.descriptions.ModelDescriptionConstants.OP_ADDR;
+import static org.wildfly.extension.messaging.activemq.MessagingServices.isSubsystemResource;
 import static org.wildfly.extension.messaging.activemq.jms.ConnectionFactoryAttributes.Common.ENTRIES;
 
 import java.util.List;
 
 import org.jboss.as.controller.AbstractRemoveStepHandler;
 import org.jboss.as.controller.OperationContext;
-import org.jboss.as.controller.PathAddress;
-import org.jboss.as.controller.descriptions.ModelDescriptionConstants;
 import org.wildfly.extension.messaging.activemq.MessagingServices;
 import org.jboss.as.naming.deployment.ContextNames;
 import org.jboss.dmr.ModelNode;
@@ -45,19 +43,19 @@ public class PooledConnectionFactoryRemove extends AbstractRemoveStepHandler {
 
     public static final PooledConnectionFactoryRemove INSTANCE = new PooledConnectionFactoryRemove();
 
-
-
+    @Override
     protected void performRuntime(OperationContext context, ModelNode operation, ModelNode model) {
-        final ModelNode operationAddress = operation.require(OP_ADDR);
-        final PathAddress address = PathAddress.pathAddress(operationAddress);
-        final String name = address.getLastElement().getValue();
-        final ServiceName serviceName = MessagingServices.getActiveMQServiceName(PathAddress.pathAddress(operation.get(ModelDescriptionConstants.OP_ADDR)));
-
-        context.removeService(JMSServices.getPooledConnectionFactoryBaseServiceName(serviceName).append(name));
-
+        ServiceName serviceName;
+        if(isSubsystemResource(context)) {
+            serviceName = MessagingServices.getActiveMQServiceName("");
+        } else {
+            serviceName = MessagingServices.getActiveMQServiceName(context.getCurrentAddress());
+        }
+        context.removeService(JMSServices.getPooledConnectionFactoryBaseServiceName(serviceName).append(context.getCurrentAddressValue()));
         removeJNDIAliases(context, model.require(ENTRIES.getName()).asList());
     }
 
+    @Override
     protected void recoverServices(OperationContext context, ModelNode operation, ModelNode model) {
         // TODO:  RE-ADD SERVICES
     }

--- a/testsuite/integration/basic/src/test/java/org/jboss/as/test/integration/messaging/mgmt/ExternalConnectionFactoryManagementTestCase.java
+++ b/testsuite/integration/basic/src/test/java/org/jboss/as/test/integration/messaging/mgmt/ExternalConnectionFactoryManagementTestCase.java
@@ -36,6 +36,7 @@ import static org.junit.Assert.assertEquals;
 import static org.junit.Assert.assertTrue;
 import static org.junit.Assert.fail;
 
+import java.io.IOException;
 import java.security.AccessController;
 import java.security.PrivilegedAction;
 
@@ -43,6 +44,8 @@ import org.jboss.arquillian.container.test.api.RunAsClient;
 import org.jboss.arquillian.junit.Arquillian;
 import org.jboss.as.arquillian.api.ContainerResource;
 import org.jboss.as.arquillian.container.ManagementClient;
+import org.jboss.as.controller.client.ModelControllerClient;
+import org.jboss.as.controller.client.helpers.Operations;
 import org.jboss.as.test.integration.common.jms.JMSOperations;
 import org.jboss.as.test.integration.common.jms.JMSOperationsProvider;
 import org.jboss.as.test.integration.management.base.ContainerResourceMgmtTestBase;
@@ -110,4 +113,39 @@ public class ExternalConnectionFactoryManagementTestCase extends ContainerResour
         ServerReload.executeReloadAndWaitForCompletion(managementClient.getControllerClient());
     }
 
+    @Test
+    public void testRemovePooledConnectionFactory() throws Exception {
+//      /subsystem=messaging-activemq/in-vm-connector=invm1:add(server-id=123)
+        ModelNode invmConnectorAddress = new ModelNode();
+        invmConnectorAddress.add("subsystem", "messaging-activemq");
+        invmConnectorAddress.add("in-vm-connector", "invm1");
+        ModelNode op = Operations.createAddOperation(invmConnectorAddress);
+        op.get("server-id").set("123");
+        execute(managementClient.getControllerClient(), op);
+
+//      /subsystem=messaging-activemq/pooled-connection-factory=pool3:add(connectors=[invm1],entries=[foo])
+        ModelNode pcfAddress = new ModelNode();
+        pcfAddress.add("subsystem", "messaging-activemq");
+        pcfAddress.add("pooled-connection-factory", "pool3");
+        op = Operations.createAddOperation(pcfAddress);
+        op.get("connectors").setEmptyList().add("invm1");
+        op.get("entries").setEmptyList().add("foo");
+        execute(managementClient.getControllerClient(), op);
+
+//      /subsystem=messaging-activemq/pooled-connection-factory=pool3:remove()
+        op = Operations.createRemoveOperation(pcfAddress);
+        execute(managementClient.getControllerClient(), op);
+
+        op = Operations.createRemoveOperation(invmConnectorAddress);
+        execute(managementClient.getControllerClient(), op);
+        ServerReload.executeReloadAndWaitForCompletion(managementClient.getControllerClient());
+    }
+
+    private static ModelNode execute(final ModelControllerClient client, final ModelNode op) throws IOException {
+        final ModelNode result = client.execute(op);
+        if (!Operations.isSuccessfulOutcome(result)) {
+            throw new RuntimeException(Operations.getFailureDescription(result).asString());
+        }
+        return result;
+    }
 }

